problem-gson/src/test/java/org/zalando/problem/GsonProblemTest.java

```diff
@@ -0,0 +1,296 @@
+import static org.hobsoft.hamcrest.compose.ComposeMatchers.hasFeature;
+import static org.zalando.problem.Status.BAD_REQUEST;
+
+class GsonProblemTest {
+
+    private static Gson gson;
+
+    @BeforeAll
+    static void init() {
+        GsonBuilder builder = new GsonBuilder();
+        // Register known sub-types
+        ProblemAdapterFactory problemAdapter = new ProblemAdapterFactory()
+                .registerSubtype(InsufficientFundsProblem.class, InsufficientFundsProblem.TYPE)
+                .registerSubtype(OutOfStockException.class, OutOfStockException.TYPE);
+        builder.registerTypeAdapterFactory(problemAdapter);
+        gson = builder.create();
+    }
+
+    private static JsonReader getReader(final String name) throws FileNotFoundException {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        URL resource = Objects.requireNonNull(loader.getResource(name), () -> "resource " + name + " not found.");
+        return new JsonReader(new FileReader(new File(resource.getPath())));
+    }
+
+    private String getStackTrace(final Throwable throwable) {
+        final StringWriter writer = new StringWriter();
+        throwable.printStackTrace(new PrintWriter(writer));
+        return writer.toString();
+    }
+
+    @Test
+    void shouldSerializeDefaultProblem() {
+        final Problem problem = Problem.valueOf(Status.NOT_FOUND);
+        final String json = gson.toJson(problem);
+
+        with(json)
+                .assertThat("$.*", hasSize(2))
+                .assertThat("$.title", is("Not Found"))
+                .assertThat("$.status", is(404));
+    }
+
+    @Test
+    void shouldSerializeCustomProperties() {
+        final Problem problem = Problem.builder()
+                .withType(URI.create("https://example.org/out-of-stock"))
+                .withTitle("Out of Stock")
+                .withStatus(BAD_REQUEST)
+                .withDetail("Item B00027Y5QG is no longer available")
+                .with("product", "B00027Y5QG")
+                .build();
+
+        final String json = gson.toJson(problem);
+
+        with(json)
+                .assertThat("$.*", hasSize(5))
+                .assertThat("$.product", is("B00027Y5QG"));
+    }
+
+    // FIXME Custom serialization not working
+    void shouldSerializeCustomProblem() {
+        final int balance = 10;
+        final int debit = 100;
+        final Problem problem = new InsufficientFundsProblem(balance, debit);
+
+        final String json = gson.toJson(problem);
+
+        with(json)
+                .assertThat("$.*", hasSize(5))
+                .assertThat("$.type", is(InsufficientFundsProblem.TYPE_VALUE))
+                .assertThat("$.title", is("Insufficient Funds"))
+                .assertThat("$.status", is(BAD_REQUEST.getStatusCode()))
+                .assertThat("$.balance", is(balance))
+                .assertThat("$.debit", is(debit));
+    }
+
+    @Test
+    void shouldSerializeProblemCause() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/whiskeysierra/problem/pullRequest?prid=4476243)